packages/sanes-chrome-extension/src/extension/background/createPersona/requestHandler.ts

```diff
@@ -0,0 +1,62 @@
+interface Request {
+  readonly reason: string;
+  readonly sender: string;
+  readonly accept: () => void;
+  readonly reject: (permanently: boolean) => void;
+}
+
+export function parseMethod(method: string): void {
+  const allowed = method === 'getIdentities' || method === 'signAndPost';
+
+  if (!allowed) {
+    throw new Error('Request method not allowed, use getIdentities or signAndPost');
+  }
+}
+
+export class RequestHandler {
+  private static instance: Request[];
```

Comment:
Why not just `private static readonly instance: Request[] = [];`? Then you don't need the create() method at all. Additionally, you can drop all those checks: `if (!RequestHandler.instance) {`

Reply:
The main reason to do not do that is because I can control in a natural way when the list is cleared, and reset it when we support switch of multiple personas.

I am not a big fan of singleton static classes to be initialised automatically. 

Do not worry about possible side effect problems because we have a test checking what happens when the RequestHandler has not `loaded`.

Reply:
Hmm, this means we combine the non-explicitness of global state with singletons and the hardness of manual initialization. Worst of both worlds ;) Anyway, fine for me as it will do the job.

> when we support switch of multiple personas.

As long as we don't support multiple Persona (which will be many months from now), there is no point at even thinking about that case. Our code will look entirely different at that point.

Closing this for now